packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -422,24 +448,32 @@ class _AppBarState extends State<AppBar> {
       ),
     );
 
-    Widget appBar = new SizedBox(
-      height: kToolbarHeight,
-      child: new IconTheme.merge(
-        context: context,
-        data: appBarIconTheme,
-        child: new DefaultTextStyle(
-          style: sideStyle,
-          child: toolbar,
+    // If the toolbar is allocated less than kToolbarHeight make it
+    // appear to scroll upwards within its shrinking container.
+    Widget appBar = new ClipRect(
+      child: new CustomSingleChildLayout(
+        delegate: new _ToolbarContainerLayout(),
```

Comment:
Can we make _ToolbarContainerLayout const constructable and use `const _ToolbarContainerLayout()` here?